File: NoodleStar/src/main/java/com/noodlestar/noodlestar/ordersubdomain/presentationlayer/OrderResponseModel.java
package com.noodlestar.noodlestar.ordersubdomain.presentationlayer;

import lombok.*;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@EqualsAndHashCode
public class OrderResponseModel {
    private String orderId;
    private String customerId;
    private String status;
    private LocalDate orderDate;
    private List<OrderDetailsResponseModel> orderDetails;
    private double total;
}

File: NoodleStar/noodlestar-frontend/src/features/model/orderResponseModel.ts
export interface OrderDetailsResponseModel {
    menuId: string;
    quantity: number;
    price: number;
}

export interface OrderResponseModel {
    orderId: string;
    customerId: string;
    status: string;
    orderDate: string;
    orderDetails: OrderDetailsResponseModel[];
    total: number;
}